src/cascadia/TerminalSettingsModel/VsSetupConfiguration.h

```diff
@@ -37,6 +37,15 @@ namespace winrt::Microsoft::Terminal::Settings::Model
     public:
         struct VsSetupInstance
         {
+            VsSetupInstance(VsSetupInstance&& other) :
+                query(std::move(other.query)),
+                inst(std::move(other.inst)),
+                profileNameSuffix(std::move(other.profileNameSuffix)),
+                installDate(other.installDate),
+                version(other.version)
+            {
+            }

```

Comment:
You can write `VsSetupInstance(VsSetupInstance&&) = default;` and move the assignment operator up to be next to it.

Reply:
That's what I had originally, but seems the default wasn't equivalent to what I have here (though it seems it should've been). It was causing dereferencing exceptions on shutdown.

Reply:
Before marking this resolved, I'll `git clean -dfx` my repo and start completely clean in case there was some issue with an incremental build (I built `main` before starting work to have a baseline). FWIW, incremental builds still took quite a while so I figured - per-project at least - they were recompiling all source and linking all new objects.

Reply:
The existing code wasn't properly AddRef'ing the `com_ptr<ISetupConfiguration2>`. Fixed, and the shutdown crash eliminated.

I'm also going to refactor this so it's only looping through instances once. It'll be more efficient for startup time, and make adding VC environments less costly.